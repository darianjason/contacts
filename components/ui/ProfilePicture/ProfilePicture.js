import React from 'react';
import {Image, View, Dimensions} from 'react-native';
import {Colors} from '../../../constants';

import {DefaultText} from '../../ui';
import styles from './ProfilePicture.styles';

const ProfilePicture = ({
  photo,
  firstName,
  lastName,
  pictureSize,
  initialsSize,
  backgroundColor,
}) => {
  if (!photo) {
    photo = 'N/A';
  }

  return (
    <View
      style={{
        ...styles.container,
        width: pictureSize ? pictureSize : Dimensions.get('window').height / 12,
        height: pictureSize
          ? pictureSize
          : Dimensions.get('window').height / 12,
        backgroundColor: backgroundColor ? backgroundColor : Colors.accent,
      }}>
      {photo !== 'N/A' ? (
        <Image
          source={{uri: photo}}
          style={styles.image}
          accessibilityRole="image"
        />
      ) : (
        <DefaultText
          style={{
            ...styles.initials,
            fontSize: initialsSize ? initialsSize : 16,
          }}
          numberOfLines={1}>
          {firstName[0]}
          {lastName[0]}
        </DefaultText>
      )}
    </View>
  );
};

export default ProfilePicture;
