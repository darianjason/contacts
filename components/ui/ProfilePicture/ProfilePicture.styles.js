import {StyleSheet} from 'react-native';

import {Colors} from '../../../constants';

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: Colors.accent,
    borderRadius: 8,
    overflow: 'hidden',
  },
  image: {
    width: '100%',
    height: '100%',
  },
  initials: {
    fontFamily: 'Manrope-Bold',
    letterSpacing: 2,
    color: Colors.primary,
  },
});

export default styles;
