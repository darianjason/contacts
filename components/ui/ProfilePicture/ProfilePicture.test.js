import React from 'react';
import {render} from '@testing-library/react-native';
import renderer from 'react-test-renderer';

import ProfilePicture from './ProfilePicture';

describe('Profile Picture component', () => {
  test('renders photo if photo is set', () => {
    const photo = 'uri';
    const firstName = 'John';
    const lastName = 'Doe';

    const {getByRole} = render(
      <ProfilePicture
        photo={photo}
        firstName={firstName}
        lastName={lastName}
      />,
    );

    expect(getByRole('image')).toBeTruthy();
  });

  test('renders initials if no photo is set', () => {
    const firstName = 'John';
    const lastName = 'Doe';

    const {getByText} = render(
      <ProfilePicture firstName={firstName} lastName={lastName} />,
    );

    expect(getByText('JD')).toBeTruthy();
  });

  test('renders correctly', () => {
    const photo = 'uri';
    const firstName = 'John';
    const lastName = 'Doe';

    const tree = renderer
      .create(
        <ProfilePicture
          photo={photo}
          firstName={firstName}
          lastName={lastName}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
