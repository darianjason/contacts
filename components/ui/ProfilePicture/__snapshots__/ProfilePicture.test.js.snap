// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Picture component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#E2EEF3",
      "borderRadius": 8,
      "height": 111.16666666666667,
      "justifyContent": "center",
      "overflow": "hidden",
      "width": 111.16666666666667,
    }
  }
>
  <Image
    accessibilityRole="image"
    source={
      Object {
        "uri": "uri",
      }
    }
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  />
</View>
`;
